apps/system/test/unit/permission_manager_test.js

```diff
@@ -93,13 +102,35 @@ suite('system/permission manager', function() {
 
   suite('permission-prompt Handler', function() {
     setup(function() {
-      this.sinon.stub(permissionManager, 'handlePermissionPrompt');
+      this.sinon.spy(permissionManager, 'handlePermissionPrompt');
+      sendChromeEvent('permission-prompt', 'test');
+    });
+
+    test('permission-prompt', function() {
+      assert.equal(permissionManager.overlay.dataset.type, 'test');
+      assert.isTrue(permissionManager.handlePermissionPrompt.called);
+    });
+  });
+
+  suite('permission-prompt queue (when requesting more than one)', function() {
+    setup(function() {
+      this.sinon.spy(permissionManager, 'handlePermissionPrompt');
+      this.sinon.spy(permissionManager, 'queuePrompt');
+      sendChromeEvent('permission-prompt', 'test');
+      sendChromeEvent('permission-prompt', 'test');
       sendChromeEvent('permission-prompt', 'test');
     });
 
     test('permission-prompt', function() {
       assert.equal(permissionManager.overlay.dataset.type, 'test');
       assert.isTrue(permissionManager.handlePermissionPrompt.called);
+      assert.isTrue(permissionManager.queuePrompt.called);
+    });
+
+    test('permission-prompt queue', function() {
+      assert.equal(permissionManager.overlay.dataset.type, 'test');
+      assert.isTrue(permissionManager.handlePermissionPrompt.called);
```

Comment:
Actually I've added a new one in order to double check that `dispatchEvent` is called the right number of times :)